qmi/core/transport.py

```diff
@@ -723,18 +738,126 @@ def discard_read(self) -> None:
         self._safe_socket.settimeout(0)
         while True:
             try:
-                b = self._safe_socket.recv(4096)
-            except socket.timeout:
+                b, addr = self._safe_socket.recvfrom(4096)
+            except (BlockingIOError, socket.timeout):
                 # no more bytes available
                 break
-            except BlockingIOError:
-                # no more bytes available
+            except OSError:
```

Comment:
this error and the next if branch need tests

Reply:
Added a test.